File: JPP/lab-1/z7/src/wrapper.c
#include "wrapper.h"

extern int  rust_factorial   (int);
extern int  rust_gcd         (int, int);
extern Pair rust_solve       (int, int, int);

int factorial(const int n)
{
    return rust_factorial(n);
}

int gcd(int a, int b)
{
    return rust_gcd(a, b);
}

Pair solve(int a, int b, int c)
{
    return rust_solve(a, b, c);
}

File: JPP/lab-1/z6/src/wrapper.rs
extern crate libc;

#[repr(C)]
pub struct Pair {
    pub x: libc::c_int,
    pub y: libc::c_int,
}

extern {
    pub fn factorial(n: libc::c_int) -> libc::c_int;
    pub fn gcd(a: libc::c_int, b: libc::c_int) -> libc::c_int;
    pub fn solve(a: libc::c_int, b: libc::c_int, c: libc::c_int) -> Pair;
}